dom/src/main/scala/org/scalajs/dom/FormData.scala

```diff
@@ -8,17 +8,112 @@ package org.scalajs.dom
 
 import scala.scalajs.js
 import scala.scalajs.js.annotation._
+import scala.scalajs.js.|
 
 /** XMLHttpRequest Level 2 adds support for the new FormData interface. FormData objects provide a way to easily
   * construct a set of key/value pairs representing form fields and their values, which can then be easily sent using
   * the XMLHttpRequest send() method.
   */
 @js.native
 @JSGlobal
-class FormData(form: HTMLFormElement = js.native) extends js.Object {
+class FormData(form: HTMLFormElement = js.native) extends js.Iterable[js.Tuple2[String, String]] {
 
-  /** Appends a key/value pair to the FormData object. */
-  def append(name: js.Any, value: js.Any, blobName: String = js.native): Unit = js.native
+  /** The `append()` method of the `FormData` interface appends a new value onto an existing key inside a `FormData`
+    * object, or adds the key if it does not already exist.
+    *
+    * @param name
+    *   The name of the field whose data is contained in value.
+    * @param value
+    *   The field's value. This can be a string or `Blob` (including subclasses such as File). If none of these are
+    *   specified the value is converted to a string.
+    */
+  def append(name: String, value: String): Unit = js.native
+
+  /** The `append()` method of the `FormData` interface appends a new value onto an existing key inside a `FormData`
+    * object, or adds the key if it does not already exist.
+    *
+    * @param name
+    *   The name of the field whose data is contained in value.
+    * @param value
+    *   The field's value. This can be a string or `Blob` (including subclasses such as File). If none of these are
+    *   specified the value is converted to a string.
+    * @param blobName
+    *   The filename reported to the server (a string), when a `Blob` or `File` is passed as the second parameter. The
+    *   default filename for `Blob` objects is "blob". The default filename for `File` objects is the file's filename.
+    */
+  def append(name: String, value: Blob, blobName: String): Unit = js.native
```

Comment:
The last argument is optional.

```suggestion
  def append(name: String, value: Blob, blobName: String = js.native): Unit = js.native
```